lib/aws-record/record/item_operations.rb

```diff
@@ -151,6 +169,10 @@ def _perform_save(opts)
             )
           end
         end
+        data = self.instance_variable_get("@data")
+        data.destroyed = false if data.destroyed
```

Comment:
Why the "if" condition here? It looks like the end condition is always going to be false or false-equivalent.